delfin/drivers/hitachi/vsp/vsp_stor.py

```diff
@@ -0,0 +1,295 @@
+            pool_list = []
+            pools = pools_info.get('data')
+            for pool in pools:
+                status = self.POOL_STATUS_MAP.get(
+                    pool.get('poolStatus'),
+                    constants.StoragePoolStatus.ABNORMAL
+                )
+                storage_type = constants.StorageType.BLOCK
+                total_cap = \
+                    int(pool.get('totalPoolCapacity')) * units.Mi
+                free_cap = int(
+                    pool.get('availableVolumeCapacity')) * units.Mi
+                used_cap = total_cap - free_cap
+                p = {
+                    'name': pool.get('poolName'),
+                    'storage_id': self.storage_id,
+                    'native_storage_pool_id': str(pool.get('poolId')),
+                    'description': 'Hitachi VSP Pool',
+                    'status': status,
+                    'storage_type': storage_type,
+                    'subscribed_capacity': int(total_cap),
+                    'total_capacity': int(total_cap),
+                    'used_capacity': int(used_cap),
+                    'free_capacity': int(free_cap),
+                }
+                pool_list.append(p)
+
+            return pool_list
+        except exception.DelfinException as err:
+            err_msg = "Failed to get pool metrics from hitachi vsp: %s" % \
+                      (six.text_type(err))
+            LOG.error(err_msg)
+            raise err
+        except Exception as e:
+            err_msg = "Failed to get pool metrics from hitachi vsp: %s" % \
+                      (six.text_type(e))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def list_volumes(self, context):
+        try:
+            volumes_info = self.rest_handler.get_all_volumes()
+
+            volume_list = []
+            volumes = volumes_info.get('data')
+            for volume in volumes:
+                if volume.get('emulationType') == 'NOT DEFINED':
+                    continue
+                orig_pool_id = volume.get('poolId')
+                compressed = False
+                deduplicated = False
+                if volume.get('dataReductionMode') == \
+                        'compression_deduplication':
+                    deduplicated = True
+                    compressed = True
+                if volume.get('dataReductionMode') == 'compression':
+                    compressed = True
+                if volume.get('status') == 'NML':
+                    status = 'normal'
+                else:
+                    status = 'abnormal'
+
+                vol_type = constants.VolumeType.THICK
+                for voltype in volume.get('attributes'):
+                    if voltype == 'HTI':
+                        vol_type = constants.VolumeType.THIN
+
+                total_cap = \
+                    int(volume.get('blockCapacity')) * consts.BLOCK_SIZE
+                used_cap = \
```

Comment:
This is going to screw the storage report. Used_cap and total_cap is same so free_cap is 0 always.

Reply:
Making free =0 may be true for a thick , because total_capacity=allocated_capacity (which is used from the perspective of array). but there should be a beeter way to update this for a thin volume. please check.

Reply:
yes,i kown that,but there is no used_capacity  or free_capacity

Reply:
So for thin devices, won't the Storage Summary report be affected by this?